addons/hr_recruitment/wizard/applicant_refuse_reason.py

```diff
@@ -29,16 +29,16 @@ class ApplicantGetRefuseReason(models.TransientModel):
     def _compute_send_mail(self):
         for wizard in self:
             template = wizard.refuse_reason_id.template_id
-            wizard.send_mail = bool(template)
+            wizard.send_mail = bool(template and not wizard.applicant_without_email)
```

Comment:
if you put an and the expression will already be converted to bool no need to try to transtype